packages/components/src/ActionList/ActionList.tsx

```diff
@@ -84,21 +85,34 @@ export interface ActionListProps {
    * Sort function provided by the developer
    */
   onSort?: (id: string, sortDirection: 'asc' | 'desc') => void
-
   /**
    * Allow the user to select ActionListItems.
    * Note: Implemented as a checkbox next to each item row.
    * @default false
    */
   canSelect?: boolean
+  /**
+   * Allow the user to select all ActinoListItems
+   * Note: Implemented as a checkbox left of the header row
+   * @default false
+   */
+  canSelectAll?: boolean
   /**
    * Callback performed when user makes a selection
    */
   onSelect?: (id: string) => void
   /**
-   * ActionListItems which should be displayed as "selected"
+   * Callback performed when user makes selects the header checkbox
+   */
+  onSelectAll?: () => void
+  /**
+   * The ids of all ActionListItems which should be displayed as "selected"
    */
   itemsSelected?: string[]
+  /**
+   * The ids of all selectable ActionListItems (i.e. all non-disabled items)
+   */
+  areAllItemsSelected?: MixedBoolean
```

Comment:
Could this be dynamically determined? Determine the IDs of all the rows and see if it equals `itemsSelected`? 

If that's not feasible I think I'd prefer a more terse prop name:

```suggestion
  /**
   * Control the state of the "Select All" checkbox
   */
  allSelected?: MixedBoolean
```

Reply:
Commit 12fa82a includes a proposed idea for dynamically determining all selected ids (that involves adding `allItems` and `setAllItems` properties to the context)

Reply:
A number of commits worked towards getting the `select-all` logic baked into the `ActionList`, but e44b2ff covers the final structure!